docs/src/references/agent/device-profile.md

```diff
@@ -172,19 +172,24 @@ on_success = "executing"
 on_error = { status = "failed", reason = "fail to sort the profile list"}
 
 [executing]
-operations = [
-    { operation = "firmware_update", target_operation="firmware_update" }
-    { operation = "software_update", target_operation="custom_software_update" }
-    { operation = "config_update" }
-]
+action = "builtin"
+on_success = "next_operation"
+
+[next_operation]
+action = "builtin"
+on_success = "apply_operation"
+on_error = { status = "failed", reason = "Failed to pick the next operation to be executed" }
+
+[apply_sub_operation]
+operations = [ "firmware_update", "software_update", "config_update" ]
```

Comment:
I guess the meaning of this list of operations. "Apply the current operation among those". Isn't it?

But what is this current operation? Can a user-provided script set it? What are the rules?

I would try to do something along the lines of what we already have:

```
operation = "${.payload.next.operation}"
```

This might require some improvements. Notably it's currently uneasy to set the initial payload of an operation in a generic way. We could extend the system with:

```
operation = "${.payload.next.operation}"
input = "${.payload.next.input}"
```

Reply:
Addressed as suggested.